thrust/cmake/README_CMake.md

```diff
@@ -0,0 +1,211 @@
+
+This option is NOT advertised in the CMake GUIs.
+
+#### `THRUST_IGNORE_CUB_VERSION_CHECK`
+
+CUB is needed to implement the CUDA backend, and the Thrust and CUB projects are
+tightly coupled. Thrust will emit an error if the CUB version does not match
+what it expects.
+
+If you are intentionally mixing Thrust and CUB versions, this check can be
+disabled by setting this option to `TRUE`.
+
+This option is NOT advertised in the CMake GUIs.
+
+#### `COMPONENTS`
+
+When calling `find_package(Thrust)`, each backend may be specified in the
+`COMPONENTS` or `OPTIONAL_COMPONENTS` options. Thrust will attempt to locate
+these backends and provide targets for them. Backends specified in the
+`THRUST_*_SYSTEM` options are automatically added as required components.
+
+Example:
+```$cmake
+find_package(Thrust REQUIRED CONFIG
+  COMPONENTS CPP CUDA
+  OPTIONAL_COMPONENTS TBB
+)
+```
+
+This will require Thrust to find and define all CPP and CUDA targets (see next
+section). Thrust will attempt to locate TBB, but will not error if it is not
+found.
+
+To test if an optional component is enabled, check for the existence of the
+corresponding `Thrust::${BACKEND}` target (in this case,
+`if (TARGET Thrust::TBB)`).
+
+## Targets Provided
+
+- `Thrust::Thrust`
+  - The main target of the Thrust project.
+  - If using the `THRUST_*_SYSTEM` options to configure Thrust, this is the only
+    target that needs to be linked.
+- `Thrust::${BACKEND}::Host` (`BACKEND` is one of `CPP`, `OMP`, `TBB`)
+  - Configures linked targets to use `BACKEND` for Thrust's host system.
+- `Thrust::${BACKEND}::Device` (`BACKEND` is one of `CUDA`, `CPP`, `OMP`, `TBB`)
+  - Configures linked targets to use `BACKEND` for Thrust's device system.
+
+The following targets should not be linked to directly from user projects. They
+may be used to detect if a backend was found (e.g. `if (TARGET Thrust::CUDA)`).
+- `Thrust::CPP` (internal)
+  - This target provides an interface to Thrust's CPP backend.
+- `Thrust::CUDA` (internal)
+  - This target provides an interface to Thrust's CUDA backend.
+- `Thrust::TBB` (internal)
+  - This target provides an interface to Thrust's TBB backend.
+- `Thrust::OMP` (internal)
+  - This target provides an interface to Thrust's OMP backend.
+
+## Debugging
+
+After Thrust configures its targets, it will print out the interface details
+of each target it defines. To suppress this output, pass the `QUIET` option to
+`find_package`.
+
+## Advanced Usage
+
+The following sections provide examples of more advanced Thrust usecases.
+
+### Specifying Thrust Version Requirements
+
+A specific version of Thrust may be required in the `find_package` call:
+
+```$cmake
+find_package(Thrust 1.9.10)
+```
+
+will only consider Thrust installations with version `1.9.10.X`. An exact match
+down to the patch version can be found using `EXACT` matching:
+
+```$cmake
+find_package(Thrust 1.9.10.3 EXACT)
+```
+
+would only match the (hypothetical) 1.9.10.3 release.
```

Comment:
Let's axe the word "(hypothetical)" here because you never know, it could happen.